### PR TITLE
Block utils

### PR DESCRIPTION
**Pull request by Seggan**
Added block utility class

### CODE REVIEW

File: src/main/java/io/github/mooy1/infinitylib/misc/BlockUtils.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+     * @param allowedMaterials all the allowed materials that can connect to the block
+     *
+     * @return a set of blocks
+     */
+    public static Set<Block> getNearbyBlocks(Block start, Set<Material> allowedMaterials) {
+        Set<Block> blocks = new HashSet<>();
+        for (int x = -1; x < 2; x++) {
+            for (int y = -1; y < 2; y++) {
+                for (int z = -1; z < 2; z++) {
+                    Block block = start.getLocation().clone().add(x, y, z).getBlock();
```
Comment (Riley31415) [inline src/main/java/io/github/mooy1/infinitylib/misc/BlockUtils.java]: dont need to clone location afaik

File: src/main/java/io/github/mooy1/infinitylib/misc/BlockUtils.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+     * This method returns a {@link Set} of blocks connected to each other, as long as those blocks' {@link Material}
+     * is in {@code allowedMaterials}
+     *
+     * @param start the block  to start from
+     * @param allowedMaterials all the allowed materials that can connect to the block
+     *
+     * @return a set of blocks
+     */
+    public static Set<Block> getNearbyBlocks(Block start, Set<Material> allowedMaterials) {
+        Set<Block> blocks = new HashSet<>();
```
Comment (Riley31415) [inline src/main/java/io/github/mooy1/infinitylib/misc/BlockUtils.java:71]: add check for allowed materials size != 0

**Review by Riley31415 — CHANGES_REQUESTED** [review]
other than that lgtm

**Comment by Seggan** [comment]
I literally grabbed the code from spigotmc forums, havent really checked over it

File: src/main/java/io/github/mooy1/infinitylib/misc/BlockUtils.java [hunk 3]
```diff
@@ -95,10 +103,10 @@
      *
      * @param start the block to start
      * @param treeType the type of the tree to get
-     *
      * @return a set of the blocks contained in the tree
      */
-    public static Set<Block> getTree(Block start, TreeType treeType) {
+    @Nonnull
+    public static Set<Block> getTree(@Nonnull Block start, @Nonnull TreeType treeType) {
```
Comment (Riley31415) [inline src/main/java/io/github/mooy1/infinitylib/misc/BlockUtils.java:110]: now this cant be get or default or itll default to all mats :notlikethis: id just do a if null return empty set

**Review by Riley31415 — CHANGES_REQUESTED** [review]

**Pull request merged**